rpcd/playbooks/roles/rpc_maas/tasks/rabbitmq_user.yml

```diff
@@ -25,3 +25,31 @@
     state: "present"
   tags:
     - rabbitmq-user
+
+  # uri module not used because httplib2 not available in the rabbit container
+- name: Get list of vhosts
+  shell: "curl -u {{ maas_rabbitmq_user }}:{{ maas_rabbitmq_password }} localhost:15672/api/vhosts"
+  register: vhost_list_output
+  tags:
+    - rabbitmq-user
+    - skip_ansible_lint
+
+- name: Parse vhost json
+  set_fact:
+    vhost_json: "{{vhost_list_output.stdout|from_json}}"
+  tags:
+    - rabbitmq-user
+
+- name: Ensure MaaS rabbitmq user as administrator
+  rabbitmq_user:
+    user: "{{ maas_rabbitmq_user }}"
+    password: "{{ maas_rabbitmq_password }}"
+    vhost: "{{item.name}}"
+    configure_priv: ".*"
+    read_priv: ".*"
+    write_priv: ".*"
+    tags: "administrator"
+    state: "present"
```

Comment:
@evrardjp, yes I don't think I explained myself very clearly, by extension I was suggesting that modifying the permissions would be similarly problematic. Looking further I see that there is a rabbitmq_user task at the top of the file and so for this task to be able to add the permissions it must already be able to modify an existing user.